### PR TITLE
Feature keeper registry 1.2.0

### PR DESCRIPTION
**Pull request by RyanRHall**

### CODE REVIEW

File: contracts/src/v0.7/dev/KeeperRegistryDev.sol [hunk 1]
```diff
@@ -337,16 +353,57 @@ contract KeeperRegistryDev is
...
+    s_expectedLinkBalance = s_expectedLinkBalance.sub(amountToWithdraw);
+    emit FundsWithdrawn(id, amountToWithdraw, to);
+
+    LINK.transfer(to, amountToWithdraw);
+  }
+
+  /**
+   * @notice withdraws LINK funds collected through cancellation fees
+   */
+  function ownerWithdrawFunds() external onlyOwner {
```
Comment (se3000) [inline contracts/src/v0.7/dev/KeeperRegistryDev.sol:385]:
```suggestion
  function withdrawOwnerFunds() external onlyOwner {
```
Naming nit-pick. From the styleguide:
> Function names should start with imperative verbs, not nouns or other tenses.

**Review by se3000 — CHANGES_REQUESTED** [review]

File: contracts/src/v0.7/dev/KeeperRegistryDev.sol (hunk 1 above)
Comment (infiloop2) [reply contracts/src/v0.7/dev/KeeperRegistryDev.sol]: fixed

**Comment by RyanRHall** [comment]
@se3000 @infiloop2 I'd like to get https://github.com/smartcontractkit/chainlink/pull/6370 merged before making further changes.

File: contracts/src/v0.8/dev/KeeperRegistryDev.sol [hunk 2]
```diff
@@ -0,0 +1,951 @@
...
+
+    uint256 id;
+    Upkeep memory upkeep;
+    uint256 totalBalanceRemaining;
+    bytes[] memory checkDatas = new bytes[](ids.length);
+    Upkeep[] memory upkeeps = new Upkeep[](ids.length);
+    for (uint256 idx = 0; idx < ids.length; idx++) {
+      id = ids[idx];
+      upkeep = s_upkeep[id];
+      if (upkeep.admin != admin) revert OnlyCallableByAdmin();
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/KeeperRegistryDev.sol:669]: The error name is confusing for readability.

**Comment by infiloop2** [comment]
Closing in favor of https://github.com/smartcontractkit/chainlink/pull/6479

**Pull request closed without merging**